app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt

```diff
@@ -867,6 +870,54 @@ class ComposeActivity :
         binding.addPollTextActionTextView.compoundDrawablesRelative[0].colorFilter = PorterDuffColorFilter(textColor, PorterDuff.Mode.SRC_IN)
     }
 
+    // Use to pass state from cropImage caller to result function
+    private var cropImageItemOld: QueuedMedia? = null
+    private var cropImageUriNew: Uri? = null
+    private var cropImageFileNew: File? = null
+
+    private val cropImage = registerForActivityResult(CropImageContract()) { result ->
+        if (result.isSuccessful) {
+            cropImageItemOld?.let { itemOld ->
+                cropImageFileNew?.let { fileNew ->
+                    cropImageUriNew?.let { uriNew ->
+                        val size = fileNew.length()
+
+                        lifecycleScope.launch {
+                            viewModel.addMediaToQueue(
+                                itemOld.type,
+                                uriNew,
+                                size,
+                                itemOld.description,
+                                itemOld
+                            )
+                        }
+                    }
+                }
+            }
+        } else {
+            // TODO: In else case check result.getError() to see if this was a voluntary cancel, display error if involuntary
+            Log.w("ComposeActivity", "Edit image aborted: " + result.error)
```

Comment:
This should be changed either in this PR or a future PR. Will ask about this in Matrix.